src/Api/JsonRpc/Client.php

```diff
@@ -422,12 +422,32 @@ private function addTestResultsToTestRun()
         }
     }
 
-    private function getCaseRunStatusIdByResult(string $result): int
+    private function getTestExecutionStatusIdByResult(string $result): int
     {
         if (isset($this->testExecutionStatus[$result])) {
             return (int) $this->testExecutionStatus[$result]['id'];
         }
 
-        return $this->testExecutionStatus['error']['id'];
+        // order test executions by weight
+        uasort($this->testExecutionStatus, function ($status1, $status2) {
+            return $status1['weight'] > $status2['weight'];
+        });
+
+        // this should not happen as BE will prevent it
+        if (strtolower($result) === 'passed') {
+            if ($this->testExecutionStatus[array_key_last($this->testExecutionStatus)]['weight'] >= 0) {
+                throw new \Exception('There is no test execution with positive weight');
+            }
+
+            // get the test execution with the highest weight
+            return (int) $this->testExecutionStatus[array_key_last($this->testExecutionStatus)]['id'];
+        }
+
+        if ($this->testExecutionStatus[array_key_first($this->testExecutionStatus)]['weight'] >= 0) {
```

Comment:
The above code block has an if condition on the status name, while this one doesn't and that's a problem b/c we want to know what kind of statuses are we looking for. 

`getTestExecutionStatusIdByResult()` is used by `addTestResult()` which in turn is used in `PHPUnitTestListener.php` in multiple functions. Each one of them tries to report a specific status that PHPUnit is able to produce and tries to map these statuses onto Kiwi TCMS' ones. 

Some of them match the names in Kiwi TCMS, others are a bit off, e.g. `FAIL` -> `FAILED`, `PASS` -> `PASSED`, `SKIPPED` -> `WAIVED`, and yet others don't exist at all, e.g. `RISKY`, `INCOMPLETE`, etc.

Once you arrive at this code block here you have lost the information what you were looking for. As written (incl the mismatches outlined above) instead of reporting PASS this implementation will report the first negative status found!

It needs some more refactoring.